### ISSUE TITLE
动态图-梯度清空

### ISSUE THREAD

**Issue by waaaaaaater** [PaddlePaddle/Paddle#30522]
在API-V1.8.0文档中，关于优化器的梯度更新时，代码如下：
~~~
import paddle.fluid as fluid
import numpy as np

with fluid.dygraph.guard():
    value = np.arange(26).reshape(2, 13).astype("float32")
    a = fluid.dygraph.to_variable(value)
    linear = fluid.Linear(13, 5, dtype="float32")
    optimizer = fluid.optimizer.SGDOptimizer(learning_rate=0.01,
                                  parameter_list=linear.parameters())
    out = linear(a)
    out.backward()
    optimizer.minimize(out)
    optimizer.clear_gradients()
~~~
这里对优化器进行了梯度清空。
而在aistudio的一些教程中，确没有进行优化器的梯度清空，仅仅对模型的梯度进行了清空。
请问是现在的版本中不需要优化器的梯度清空，优化器的梯度不是累加的吗？
不同的教程中有的时候写的是优化器梯度需要清零，有的则不写。
有的时候写得网络的梯度需要清零，有的则不写？
是不同版本，内部接口处理不一样吗？
~~~
with fluid.dygraph.guard():
    model=MyDNN() #模型实例化
    model.train() #训练模式
    opt=fluid.optimizer.SGDOptimizer(learning_rate=train_parameters['learning_strategy']['lr'], parameter_list=model.parameters())#优化器选用SGD随机梯度下降，学习率为0.001.
    epochs_num=train_parameters['num_epochs'] #迭代次数
    
    for pass_num in range(epochs_num):
        for batch_id,data in enumerate(train_reader()):
            images=np.array([x[0].reshape(1,20,20) for x in data],np.float32)
            labels = np.array([x[1] for x in data]).astype('int64')
            labels = labels[:, np.newaxis]
            image=fluid.dygraph.to_variable(images)
            label=fluid.dygraph.to_variable(labels)

            predict=model(image) #数据传入model
            
            loss=fluid.layers.cross_entropy(predict,label)
            avg_loss=fluid.layers.mean(loss)#获取loss值
            
            acc=fluid.layers.accuracy(predict,label)#计算精度
            
            if batch_id!=0 and batch_id%50==0:
                Batch = Batch+50 
                Batchs.append(Batch)
                all_train_loss.append(avg_loss.numpy()[0])
                all_train_accs.append(acc.numpy()[0])
                
                print("train_pass:{},batch_id:{},train_loss:{},train_acc:{}".format(pass_num,batch_id,avg_loss.numpy(),acc.numpy()))
            
            avg_loss.backward()       
            opt.minimize(avg_loss)    #优化器对象的minimize方法对参数进行更新 
            model.clear_gradients()   #model.clear_gradients()来重置梯度
    fluid.save_dygraph(model.state_dict(),'MyDNN')#保存模型

draw_train_acc(Batchs,all_train_accs)
draw_train_loss(Batchs,all_train_loss)

~~~

**Comment by zhwesky2010** [comment]
@waaaaaaater 您好，现在的paddle版本都需要进行梯度清空的，模型或者优化器梯度清空的效果是一致的

**Comment by waaaaaaater** [comment]
@zhouwei25  请问是
随便清空模型或优化器的梯度，就可以达到清空的效果吗？

**Comment by zhwesky2010** [comment]
@waaaaaaater 对，都是一样的，清理的都是相应参数的梯度，optimizer有个param_list参数，清理的是这个里面的参数

**Comment by waaaaaaater** [comment]
@zhouwei25  好的 谢谢！
